src/s2_fos/training/open_ai_prompts.py

```diff
@@ -0,0 +1,209 @@
+                else:
+                    open_ai_calls[key] = {'prompt': messages, 'result': result}
+                success = True
+            except openai.OpenAIError as e:
+                print(f"Error: {e}")
+                if messages:
+                    print(f'Length of the message: {len(tokenize(json.dumps(messages)))}')
+                if truncated_title and truncated_abstract:
+                    print(
+                        f'Length of title: {len(tokenize(truncated_title))} Length of Abstract: {len(tokenize(truncated_abstract))}')
+                print(
+                    f'Paper_sha: {paper_sha} Title: {truncated_title}\n Abstract: {truncated_abstract}\n Journal Name: {journal_name}')
+                if retries < max_retries:
+                    print(f"Retrying in {retry_delay} seconds...")
+                    time.sleep(retry_delay)
+                    retries += 1
+                else:
+                    print("Max retries reached. Skipping this API call.")
+    return open_ai_calls
+
+
+def prompt_with_journal(title, abstract, journal_name):
+    message = [
+        {"role": "system", "content": "You are a highly intelligent and accurate information extraction system. You take title, abstract, journal name of a \
+         scientific article as input and your task is to classify the scientific field of study of the passage.",
+         "role": "user", "content": "You need to classify it with key: 'field_of_study' assign as many 'field_of_study' as you find it fit: \
+    'Agricultural and Food sciences', \
+    'Art', \
+    'Biology',\
+    'Business',\
+    'Chemistry',\
+    'Computer science',\
+    'Economics',\
+    'Education',\
+    'Engineering',\
+    'Environmental science',\
+    'Geography',\
+    'Geology',\
+    'History',\
+    'Law',\
+    'Linguistics',\
+    'Materials science',\
+    'Mathematics',\
+    'Medicine',\
+    'Philosophy',\
+    'Physics',\
+    'Political science',\
+    'Psychology',\
+    'Sociology'\
+    Only select from the above list, or 'Other'."},
+
+        {"role": "assistant",
+         "content": ("```python \n"
+                     f"title = {title} \n"
+                     f"abstract = {abstract}\n"
+                     f"journal_name = {journal_name}\n"
+                     "{'field_of_study': ["
+                     )}, ]
+    return message
+
+
+idx = 0
+model = "gpt-3.5-turbo"
+max_retries = 5
+retry_delay = 2  # Time in seconds to wait before retrying
+
+meta_data_np = np.array()
+for paper_sha, title, abstract, journal_name in meta_data_np:
+
+    open_ai_calls = process_paper(paper_sha, title, abstract, journal_name, model, open_ai_calls)
+    idx += 1
+    if idx % 100 == 0:
+        with open(f'../upsample_under_perform_s2_fos_2/open_ai_most_popular_{idx}.pkl', 'wb') as file_handle:
+            pickle.dump(open_ai_calls, file_handle)
+
+with open(f'../upsample_under_perform_s2_fos_2/open_ai_most_popular_final.pkl', 'wb') as file_handle:
```

Comment:
Are we including these pkl in the release? No need for people to rerun all this... Let's include.

Reply:
Will do, will add it to the hugging face data part

Reply:
It's good practice to add everywhere with relative imports like in readme